test/python/utils/mitigation/test_meas.py

```diff
@@ -0,0 +1,707 @@
+
+        # Apply the calibration matrix to results
+        # in list and dict forms using different methods
+        results_dict_1 = meas_filter.apply(
+            results_dict, method="least_squares", meas_layout=meas_layout
+        )
+        results_dict_0 = meas_filter.apply(
+            results_dict, method="pseudo_inverse", meas_layout=meas_layout
+        )
+
+        # Assert that the results are equally distributed
+        self.assertDictEqual(results_dict, results_dict_0)
+        round_results = {}
+        for key, val in results_dict_1.items():
+            round_results[key] = np.round(val)
+        self.assertDictEqual(results_dict, round_results)
+
+    def test_tensored_meas_cal_on_circuit(self):
+        """Test an execution on a circuit."""
+
+        # Generate the calibration circuits
+        meas_calibs, mit_pattern, ghz, meas_layout = tensored_calib_circ_creation()
+
+        # Run the calibration circuits
+        backend = Aer.get_backend("qasm_simulator")
+        cal_results = qiskit.execute(
+            meas_calibs,
+            backend=backend,
+            shots=self.shots,
+            seed_simulator=SEED,
+            seed_transpiler=SEED,
+        ).result()
+
+        # Make a calibration matrix
+        meas_cal = TensoredMeasFitter(cal_results, mit_pattern=mit_pattern)
+        # Calculate the fidelity
+        fidelity = meas_cal.readout_fidelity(0) * meas_cal.readout_fidelity(1)
+
+        results = qiskit.execute(
+            [ghz], backend=backend, shots=self.shots, seed_simulator=SEED, seed_transpiler=SEED
+        ).result()
+
+        # Predicted equally distributed results
+        predicted_results = {"000": 0.5, "111": 0.5}
+
+        meas_filter = meas_cal.filter
+
+        # Calculate the results after mitigation
+        output_results_pseudo_inverse = meas_filter.apply(
+            results, method="pseudo_inverse", meas_layout=meas_layout
+        ).get_counts(0)
+        output_results_least_square = meas_filter.apply(
+            results, method="least_squares", meas_layout=meas_layout
+        ).get_counts(0)
+
+        # Compare with expected fidelity and expected results
+        self.assertAlmostEqual(fidelity, 1.0)
+        self.assertAlmostEqual(
+            output_results_pseudo_inverse["000"] / self.shots, predicted_results["000"], places=1
+        )
+
+        self.assertAlmostEqual(
+            output_results_least_square["000"] / self.shots, predicted_results["000"], places=1
+        )
+
+        self.assertAlmostEqual(
+            output_results_pseudo_inverse["111"] / self.shots, predicted_results["111"], places=1
+        )
+
+        self.assertAlmostEqual(
+            output_results_least_square["111"] / self.shots, predicted_results["111"], places=1
+        )
+
+
+@unittest.skipUnless(HAS_AER, "Qiskit aer is required to run these tests")
+class TestMeasCalNoise(TestMeasCal):
+    """Run measurement mitigation calibration tests with noise."""
```

Comment:
This inheritance causes `TestMeasCalNoise` to duplicate the previous five tests, whereas I suspect you were only trying to inherit the non-test methods?  Is there anything stopping these tests just being in the same class, since they logically are pretty linked?

Reply:
No they probably can be in the same class, I was trying to model the different files from ignis into a single file with different classes corresponding to the grouping of tests in ignis (IIRC things were split based on the use of json input files). But there is no reason to split them here I'll remove the subclassing.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/6867/commits/5a6070f4abec2f9ef536b685a829934f81c34110